numba/tests/test_parfors.py

```diff
@@ -1557,6 +1557,21 @@ def test_impl(a, b):
                                     (types.Array(types.float64, 1, 'C'),
                                      types.Array(types.float64, 1, 'C'))) == 1)
 
+    @skip_parfors_unsupported
+    def test_tuple_concat(self):
+        def test_impl(a):
+            n = len(a)
+            array_shape = n, n
+            indices = np.zeros((1,) + array_shape, dtype=np.uint64)
```

Comment:
Would the following pass?
```suggestion
            indices = np.zeros((1,) + array_shape + (1,), dtype=np.uint64)
```
?

Reply:
The first should work but I doubt the second.

Reply:
There was a small bug with the first one that I've fixed.  I've also enabled the second use case as well. @eric-wieser